### PR TITLE
feat: mark off connection state and response state

### LINKED ISSUE #53: detect agent socket event
Make ConnectionTimeout meaning timeout on connection...

### PR DESCRIPTION
**Pull request by fengmk2**
use req socket event to impl this

Fixes #53

### CODE REVIEW

File: test/urllib.test.js [hunk 1]
```diff
@@ -522,6 +531,9 @@ describe('urllib.test.js', function () {
                 console.log(res.statusCode, res.headers);
               }
               res.should.have.header('connection', 'keep-alive');
+              if (index >=3) {
```
Comment (dead-horse) [inline test/urllib.test.js]: space

Comment (fengmk2) [inline test/urllib.test.js]: 嗯，我等会再提交一下测试修改，让测试更加稳定

**Comment by fengmk2** [comment]
好了，测试应该稳定下来了

**Comment by dead-horse** [comment]
:100:

**Comment by fengmk2** [comment]
urllib@2.3.0


Original Message
Sender:Yiyu Henotifications@github.com
Recipient:node-modules/urlliburllib@noreply.github.com
Cc:fengmk2m@fengmk2.com
Date:Monday, Feb 16, 2015 14:02
Subject:Re: [urllib] feat: mark off connection state and response state (#54)


Merged #54.

**Pull request merged**